server/src/services/commits.ts

```diff
@@ -14,6 +14,51 @@
  * limitations under the License.
  */
 
+import {
+  Filter,
+  CommitResponse,
+  QueryParams,
+  CommitPayloadKey,
+} from '../interfaces';
 import {commitStorage} from '../storage';
 
-export const commitService = {};
+/**
+ * Retrieves all commits.
+ * If query paramaters are provided, will retrieve all commits satisfying
+ * the query parameters.
+ * @param queryParams The query parameters
+ */
+const getAllCommits = async (
+  queryParams?: QueryParams
+): Promise<CommitResponse[]> => {
+  if (queryParams === undefined) {
+    return commitStorage.getAllCommits();
```

Comment:
is there a valid use case for this?

Reply:
@ifhuang  yea I actually don't think there is a need to get all commits anymore HAHA shall return 404 instead if no query params are fed in